Failed to build PDF with xelatex

### Problem
- Failed to build PDF document using xelatex

#### Procedure to reproduce the problem
```
$ sphinx-quickstart doc
$ cd doc/
$ echo 'latex_engine = "xelatex"' >> conf.py
$ make latexpdf
Running Sphinx v1.6+
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [latex]: all documents
updating environment: 1 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
processing test.tex...index
resolving references...
writing... done
copying TeX support files...
done
build succeeded.
latexmk -pdfxe -dvi- -ps-  'test.tex'
Latexmk: -pdfxe bad option

Latexmk: Bad options specified
Use
   latexmk -help
to get usage information
make[1]: *** [test.pdf] Error 10
```

#### Expected results
- Succeeded to build PDF.

### Environment info
- OS: Mac
- Python version: 2.7 / 3.6
- Sphinx version: HEAD of 1.6-release branch
- MacTeX 20161009 (installed by homebrew-cask)

I also got similar error for `latex_engine = "lualatex"`.
```
latexmk -pdflua -dvi- -ps-  'test.tex'
Latexmk: -pdflua bad option

Latexmk: Bad options specified
Use
   latexmk -help
to get usage information
make[1]: *** [test.pdf] Error 10
```
And also fixed by replacing `-pdflua` option by `-lualatex`.

@jfbu  In `sphinx/texinputs/Makefile_t`, the options strings of `LATEX` and `PDFLATEX` are different. Is this intended?
```
{% elif latex_engine == 'lualatex' -%}
LATEX = latexmk -lualatex
PDFLATEX = latexmk -pdflua -dvi- -ps-
{% elif latex_engine == 'xelatex' -%}
LATEX = latexmk -pdfxe -dvi- -ps-
PDFLATEX = $(LATEX)
{% endif %}
```

--- a/CHANGES
+++ b/CHANGES
@@ -28,6 +28,8 @@ Bugs fixed
 * #3725: Todo looks different from note in LaTeX output
 * #3479: stub-columns have no effect in LaTeX output
 * #3738: Nonsensical code in theming.py
+* #3746: PDF builds fail with latexmk 4.48 or earlier due to undefined
+  options ``-pdfxe`` and ``-pdflua``
 
 Testing
 --------

--- a/doc/builders.rst
+++ b/doc/builders.rst
@@ -211,8 +211,12 @@ The builder's "name" must be given to the **-b** command-line option of
 
          make latexpdf LATEXMKOPTS="-silent"
 
-      reduces console output to a minimum. To pass options directly to the
-      ``pdflatex`` executable, use variable ``LATEXOPTS`` (for example
+      reduces console output to a minimum. Also, if ``latexmk`` version is
+      4.52b or higher (Jan 17) and ``xelatex`` is the :confval:`latex_engine`,
+      then ``LATEXMKOPTS="-xelatex"`` will speed up PDF builds.
+
+      To pass options directly to the
+      ``(pdf|xe|lua)latex`` executable, use variable ``LATEXOPTS`` (for example
       ``LATEXOPTS="--interaction=nonstopmode"``).
 
    .. autoattribute:: name

--- a/sphinx/texinputs/Makefile_t
+++ b/sphinx/texinputs/Makefile_t
@@ -17,23 +17,21 @@ ARCHIVEPRREFIX =
 # Additional LaTeX options (passed via variables in latexmkrc/latexmkjarc file)
 export LATEXOPTS =
 # Additional latexmk options
+{% if latex_engine == 'xelatex' -%}
+# with latexmk version 4.52b or higher set LATEXMKOPTS to -xelatex either here
+# or on command line for faster builds.
+{% endif -%}
 LATEXMKOPTS =
-# format: pdf or dvi
+# format: pdf or dvi (used only by archive targets)
 FMT = pdf
 
 {% if latex_engine == 'platex' -%}
 # latexmkrc is read then overridden by latexmkjarc
 LATEX = latexmk -r latexmkjarc -dvi
 PDFLATEX = latexmk -r latexmkjarc -pdfdvi -dvi- -ps-
-{% elif latex_engine == 'pdflatex' -%}
+{% else -%}
 LATEX = latexmk -dvi
 PDFLATEX = latexmk -pdf -dvi- -ps-
-{% elif latex_engine == 'lualatex' -%}
-LATEX = latexmk -lualatex
-PDFLATEX = latexmk -pdflua -dvi- -ps-
-{% elif latex_engine == 'xelatex' -%}
-LATEX = latexmk -pdfxe -dvi- -ps-
-PDFLATEX = $(LATEX)
 {% endif %}
 
 %.png %.gif %.jpg %.jpeg: FORCE_MAKE

--- a/sphinx/texinputs/latexmkrc_t
+++ b/sphinx/texinputs/latexmkrc_t
@@ -1,5 +1,13 @@
+{% if latex_engine == 'pdflatex' -%}
 $latex = 'latex ' . $ENV{'LATEXOPTS'} . ' %O %S';
 $pdflatex = 'pdflatex ' . $ENV{'LATEXOPTS'} . ' %O %S';
+{% elif latex_engine == 'lualatex' -%}
+$latex = 'lualatex --output-format=dvi ' . $ENV{'LATEXOPTS'} . ' %O %S';
+$pdflatex = 'lualatex ' . $ENV{'LATEXOPTS'} . ' %O %S';
+{% elif latex_engine == 'xelatex' -%}
+$latex = 'xelatex --no-pdf ' . $ENV{'LATEXOPTS'} . ' %O %S';
+$pdflatex = 'xelatex ' . $ENV{'LATEXOPTS'} . ' %O %S';
+{% endif -%}
 $lualatex = 'lualatex ' . $ENV{'LATEXOPTS'} . ' %O %S';
 $xelatex = 'xelatex --no-pdf ' . $ENV{'LATEXOPTS'} . ' %O %S';
 $makeindex = 'makeindex -s python.ist %O -o %D %S';

